### PR TITLE
[Core][ModelPartIO] Fix flag writing

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

When looking at the comments from @roigcarlo in #10157 I realized that the way to write the flags in ModelPartIO is not correct (I wrote it 6 years ago, completely normal) 

**🆕 Changelog**

- [Refactor flag write](https://github.com/KratosMultiphysics/Kratos/commit/cfba7950c6cb347aaa3b9112abaae9f1a6d9369f)
- [Typo](https://github.com/KratosMultiphysics/Kratos/commit/868979d4301cd1ccef59937c21cced60fde06822)
- [Clean up components variable](https://github.com/KratosMultiphysics/Kratos/commit/0ff393b66f883ffff0265c65b69bcc50f15cfc6c)
- [Style correction](https://github.com/KratosMultiphysics/Kratos/commit/aafcaf5ff975790112b2243d53f8364cf48935c9)
- [Testing reading flags](https://github.com/KratosMultiphysics/Kratos/commit/1c70140c4d3a26a379affb2cd9368715cc950314)
- [Testing writing flags](https://github.com/KratosMultiphysics/Kratos/commit/b9979e67a31b1724a549cdc7d678916055ea1793)
- [Reduce code duplication from former components (now Variable<double>)](https://github.com/KratosMultiphysics/Kratos/pull/10160/commits/8ad2eb74b05968006881c4693c91df6e9941186d)

### CODE REVIEW

**Comment by loumalouomega** [comment]
Ping @roigcarlo

File: kratos/sources/model_part_io.cpp [hunk 1]
```diff
@@ -2123,119 +2112,101 @@ void ModelPartIO::WriteNodalDataBlock(ModelPart& rThisModelPart)
...
+        if (to_consider == 1) {
+            (*mpStream) << "Begin NodalData\t" << r_flag_name << std::endl;
+            for(std::size_t j = 0; j < number_of_nodes; j++) {
+                auto it_node = it_node_begin + j;
+                if (it_node->Is(r_variable_flag)) {
+                    (*mpStream) << it_node->Id() <<"\n";
+                }
+            }
+            (*mpStream) << "End NodalData" << std::endl << std::endl;
+        }
```
Comment (roigcarlo) [inline kratos/sources/model_part_io.cpp:2144]: This is O(n²) in disguise wee need to think another way to do it.

**Review by roigcarlo — COMMENTED** [review]
I see no problems in the code but the performance impact here could be potentially huge, let me think a little more about it.

File: kratos/sources/model_part_io.cpp (hunk 1 above)
Comment (roigcarlo) [reply kratos/sources/model_part_io.cpp:2144]: Wait, is 2n, my bad... ok

**Review by roigcarlo — APPROVED** [review]

**Pull request merged**